rust/src/jitter.rs

```diff
@@ -0,0 +1,375 @@
+                            let referrer_info = ReferrerInfo::get_referrer_info(taker_stats);
+
+                            let spot_params = self.spot_params.clone();
+                            let ongoing_auction = tokio::spawn(async move {
+                                if let Some(param) = spot_params.get(&order.market_index) {
+                                    let _ = jitter.try_fill(
+                                        user.clone(), 
+                                        Pubkey::from_str(&taker).unwrap(), 
+                                        user_stats_key.clone(), 
+                                        order.clone(), 
+                                        order_signature.clone(),
+                                        referrer_info,
+                                        param.clone()
+                                    ).await;
+                                };
+                            });
+
+                            self.ongoing_auctions.insert(order_sig, ongoing_auction);
+                        } else {
+                            log::warn!("Jitter not listening to {}", order.market_index);
+                            return Ok(())
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    // Helper functions
+    pub fn update_perp_params(&self, market_index: u16, params: JitParams) {
+        self.perp_params.insert(market_index, params);
+    }
+
+    pub fn update_spot_params(&self, market_index: u16, params: JitParams) {
+        self.spot_params.insert(market_index, params);
+    }
+
+    pub fn set_user_filter(&mut self, user_filter: Box<UserFilter>) {
+        self.user_filter = Some(user_filter);
+    }
+
+    pub fn get_order_signatures(&self, taker_key: &str, order_id: u32) -> String {
+        format!("{}-{}", taker_key, order_id)
+    }
+}
+
+pub struct Shotgun<T: AccountProvider> {
+    pub jit_proxy_client: JitProxyClient<T>
+}
+
+/// Implementing the Sniper is left as an exercise for the reader.
+/// The Sniper is a more patient and precise creature, waiting for the perfect moment to strike.
+/// It is also a more complex mechanism, necessitating a careful hand to design properly.
+/// The Sniper is an advanced strategy, and is not for the faint of heart.
+/// Good luck!
+pub struct Sniper<T: AccountProvider> {
+    pub jit_proxy_client: JitProxyClient<T>,
+    pub slot_subscriber: SlotSubscriber
+}
+
+#[async_trait]
+impl<T: AccountProvider> JitterStrategy for Shotgun<T> {
+    async fn try_fill(
+        &self, 
+        taker: User, 
+        taker_key: Pubkey, 
+        taker_stats_key: Pubkey, 
+        order: Order, 
+        order_sig: String,
+        referrer_info: Option<ReferrerInfo>,
+        params: JitParams,
+    ) -> JitResult<()> {
+        log::info!("Trying to fill with Shotgun:");
+        log_details(&order);
+        
+        for i in 0..order.auction_duration {
```

Comment:
kinda nit, but this seems kinda arbitrary since there isn't any time component in this loop, so you'll end up just sending 20 txs within the first 2 slots of the auction.

Just some suggestions (feel free to ignore):
* sleep a bit between each send
* re-trigger try_fill on a slot update
* send any arbitary number of txs (if ur gonna send 20, why not 200?)